x-pack/plugins/cloud/public/custom_events_system.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { customEvents } from '@kbn/custom-events';
+import { CoreSetup, IBasePath, PackageInfo, PluginInitializerContext } from 'kibana/public';
+import { Subscription } from 'rxjs';
+import {
+  AuthenticatedUser,
+  AuthenticationServiceStart,
+} from '../../../../x-pack/plugins/security/public';
+import { CloudConfigType } from '.';
+
+/**
+ * This is the entry point used to boot the frontend when serving a application
+ * that lives in the Kibana Platform.
+ */
+
+interface SetupDeps {
+  authc?: AuthenticationServiceStart;
+  esOrgId: string;
+}
+
+export class CustomEventsSystem {
+  private readonly enabled?: boolean;
+  private basePath!: IBasePath;
+  private appSubscription?: Subscription;
+  private packageInfo: Readonly<PackageInfo>;
+  /**
+   * `apmConfig` would be populated with relevant APM RUM agent
+   * configuration if server is started with elastic.apm.* config.
+   */
+  constructor(
+    initializerContext: PluginInitializerContext,
+    private readonly config: CloudConfigType
+  ) {
+    this.packageInfo = initializerContext.env.packageInfo;
+    this.enabled = config.full_story?.enabled;
+  }
+
+  async setup(coreSetup: CoreSetup, deps: SetupDeps) {
```

Comment:
NIT: We are not awaiting any promise inside this method. I think it's safe to remove the `async` part :)